src/dbnode/encoding/types.go

```diff
@@ -24,19 +24,26 @@ import (
 	"io"
 	"time"
 
+	"github.com/m3db/m3/src/dbnode/storage/namespace"
 	"github.com/m3db/m3/src/dbnode/ts"
 	"github.com/m3db/m3/src/dbnode/x/xio"
 	"github.com/m3db/m3/src/dbnode/x/xpool"
-	"github.com/m3db/m3/src/x/serialize"
 	"github.com/m3db/m3/src/x/checked"
 	"github.com/m3db/m3/src/x/ident"
 	"github.com/m3db/m3/src/x/pool"
+	"github.com/m3db/m3/src/x/serialize"
 	xtime "github.com/m3db/m3/src/x/time"
 )
 
 // Encoder is the generic interface for different types of encoders.
 type Encoder interface {
+	// SetSchema sets up the schema needed by schema-aware encoder to encode the stream.
+	// SetSchema can be called multiple times between reset for mid-stream schema changes.
+	SetSchema(descr namespace.SchemaDescr)
```

Comment:
nit: Do you think people need to SetSchema(...) in between Reset(...) calls?  

If not is it worth removing if there's no legitimate real use case?  Otherwise keeping it I suppose makes sense.

Reply:
This is for mid-stream schema change case.

Reply:
I see, makes sense.  Please do ignore then =]